crt_rpc.c

```diff
@@ -0,0 +1,1643 @@
+/*
+ * (C) Copyright 2016-2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is part of CaRT. It implements the main RPC routines.
+ */
+#define D_LOGFAC	DD_FAC(rpc)
+
+#include "crt_internal.h"
+
+#define CRT_CTL_MAX_LOG_MSG_SIZE 256
+
+int
+crt_req_set_retry(struct crt_rpc_priv* rpc_priv) {
+        return (atomic_fetch_or_relaxed(&rpc_priv->crp_rpc_retry, CRT_RPC_RETRY) & ~CRT_RPC_RETRY);
```

Comment:
(style)  line over 80 characters
(style)  code indent should use tabs where possible